app/_src/using-mesh/managing-ingress-traffic/builtin-k8s.md

```diff
@@ -32,9 +33,34 @@ spec:
   tags:
     kuma.io/service: edge-gateway
 ```
+{% endif_version %}
+{% if_version gte:2.7.x %}
+{% if_version lte:2.9.x %}
+[//]: # (This is change in behavior, let's assume that users will get used to it, so we won't have to show this warning after 2.9.x)
+{% warning %}
+**Heads up!**
+In previous versions of {{site.mesh_product_name}}, setting the `kuma.io/service` tag directly within a `MeshGatewayInstance` resource was used to identify the service. However, this practice is deprecated and no longer recommended for security reasons since {{site.mesh_product_name}} version 2.7.0.
 
-Once a `MeshGateway` exists with `kuma.io/service: edge-gateway`, the control plane creates a new `Deployment` in the `default` namespace.
-This `Deployment` deploys 2 replicas of `kuma-dp` and corresponding builtin gateway `Dataplane` running with `kuma.io/service: edge-gateway`.
+We've automatically switched to generating the service name for you based on your `MeshGatewayInstance` resource name and namespace (format: `{name}_{namespace}_svc`).
+{% endwarning %}
+
+{% endif_version %}
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshGatewayInstance
+metadata:
+  name: edge-gateway
+  namespace: default
+  labels:
+    kuma.io/mesh: default # only necessary if not using default Mesh
+spec:
+  replicas: 2
+  serviceType: LoadBalancer
+```
+{% endif_version %}
+
+Once a `MeshGateway` exists with {% if_version lte:2.6.x inline:true %}`kuma.io/service: edge-gateway`{% endif_version %}{% if_version gte:2.7.x inline:true %}`kuma.io/service: edge-gateway_default_svc`{% endif_version %}, the control plane creates a new `Deployment` in the `default` namespace.
+This `Deployment` deploys 2 replicas of `kuma-dp` and corresponding builtin gateway `Dataplane` running with {% if_version lte:2.6.x inline:true %}`kuma.io/service: edge-gateway`{% endif_version %}{% if_version gte:2.7.x inline:true %}`kuma.io/service: edge-gateway_default_svc`{% endif_version %}.
```

Comment:
Should we maybe copy whole sentences? This is hard to read

Reply:
done